src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.Split.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System
+{
+    public static partial class MemoryExtensions
+    {
+        public static SpanSplitEnumerator<char> Split(this ReadOnlySpan<char> span)
+            => new SpanSplitEnumerator<char>(span, ' ');
+
+        public static SpanSplitEnumerator<char> Split(this ReadOnlySpan<char> span, char separator)
+            => new SpanSplitEnumerator<char>(span, separator);
+
+        public static SpanSplitEnumerator<char> Split(this ReadOnlySpan<char> span, string separator)
+            => new SpanSplitEnumerator<char>(span, separator);
+    }
+
+    public ref struct SpanSplitEnumerator<T> where T : IEquatable<T>
+    {
+        private readonly ReadOnlySpan<T> _sequence;
+        private readonly ReadOnlySpan<T> _separators;
+        private readonly T _separator;
+        private readonly bool _isSequence;
+        private readonly int _separatorLength;
+        private int _offset;
+        private int _index;
+
+        public SpanSplitEnumerator<T> GetEnumerator() => this;
+        public readonly Range Current => new Range(_offset, _offset + _index - _separatorLength);
+
+        internal SpanSplitEnumerator(ReadOnlySpan<T> span, ReadOnlySpan<T> separators)
+        {
+            _sequence = span;
+            _separators = separators;
+            _separator = default;
```

Comment:
Because of Nullable Reference Types this line results in an error, as default(T) will be null in the case of T is class.
I feel like making `_separator` nullable would just result in an unnecessary null-check in the MoveNext() method. How should I proceed with this? `#nullable disable` perhaps?